lib/core/collection/hash_collection.nit

```diff
@@ -272,6 +272,12 @@ class HashMap[K, V]
 		enlarge(0)
 	end
 
+	# Build a list filled with the items of `coll`.
+	init from(coll: Map[K, V]) do
+		init
+		for k, v in coll do self[k] = v
+	end
```

Comment:
`Map::recover_with` does the loop part.

https://github.com/nitlang/nit/blob/master/lib/core/collection/abstract_collection.nit#L677